common/Scripts/worklog.py

```diff
@@ -121,37 +159,75 @@ def cves(self) -> Set[str]:
     def ghsas(self) -> Set[str]:
         return {m[0] for m in re.findall(r'(GHSA(-[23456789cfghjmpqrvwx]{4}){3})', self.message)}
 
+
+class GitHubPR(Listable):
     @staticmethod
-    def __tempfile(ref: str) -> str:
-        dir = os.path.join(tempfile.gettempdir(), '_solus_worklog')
-        os.makedirs(dir, exist_ok=True, mode=0o700)
+    def get(id: int) -> 'GitHubPR':
+        return GitHubPR(GitHubObject.get(f'pr_{id}', f'repos/getsolus/packages/pulls/{id}'))
 
-        return os.path.join(dir, ref + '.json')
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self._data = data
 
-    @staticmethod
-    def __get_cache(ref: str) -> Optional['GitHubCommit']:
-        file = GitHubCommit.__tempfile(ref)
-        if not os.path.exists(file):
-            return None
+    @property
+    def package(self) -> str:
+        if ':' not in self.title:
+            return '<unknown>'
 
-        with open(file, 'rb') as f:
-            return GitHubCommit(json.loads(f.read()))
+        return self.title.split(':', 2)[0].strip()
 
-    @staticmethod
-    def __store_cache(ref: str, commit: str) -> None:
-        with open(os.path.join(GitHubCommit.__tempfile(ref)), 'w') as fh:
-            fh.write(commit)
+    @property
+    def title(self) -> str:
+        return self._data['title']  # type: ignore
 
-    @staticmethod
-    def __get_gh_cli(ref: str) -> 'GitHubCommit':
-        res = subprocess.run(['gh', 'api', f'repos/getsolus/packages/commits/{ref}'],
-                             capture_output=True, text=True)
-        if res.returncode != 0:
-            raise ValueError(f'GitHub API returned non-zero exit: {res.stderr}')
+    @property
+    def date(self) -> datetime:
+        return datetime.fromisoformat(self._data['merged_at'])
+
+    @property
+    def body(self) -> str:
+        return str(self._data['body'])
+
+    @property
+    def summary(self) -> str:
+        start = self.body.find('**Summary**')
+        end = self.body.find('**Test Plan**')
+
+        if start < 0:
+            return self.body
+
+        return self.body[start + 11:end].strip()
 
-        GitHubCommit.__store_cache(ref, res.stdout)
+    @property
+    def labels(self) -> list[str]:
+        return [label['name'] for label in self._data['labels']]
 
-        return GitHubCommit(json.loads(res.stdout))
+    @property
+    def include_in_sync_notes(self) -> bool:
+        return 'Topic: Sync Notes' in self.labels
+
+    def to_md(self) -> str:
+        return f'[{self._list_title()}]({self._url}\n{self._prefix_summary()}\n'
```

Comment:
This is missing a closing `)` for the URL link.

Reply:
Whoops, fixed!